components/eventing-controller/pkg/backend/jetstreamv2/utils.go

```diff
@@ -0,0 +1,222 @@
+			CleanType:    cleanType,
+		}
+		cleanEventTypes = append(cleanEventTypes, newEventType)
+	}
+	return cleanEventTypes, nil
+}
+
+// isJsSubAssociatedWithKymaSub returns true if the given SubscriptionSubjectIdentifier and Kyma subscription
+// have the same namespaced name, otherwise returns false.
+func isJsSubAssociatedWithKymaSub(jsSubKey SubscriptionSubjectIdentifier, subscription *eventingv1alpha2.Subscription) bool {
+	return createKeyPrefix(subscription) == jsSubKey.NamespacedName()
+}
+
+//----------------------------------------
+// SubscriptionSubjectIdentifier utils
+//----------------------------------------
+
+// NamespacedName returns the Kubernetes namespaced name.
+func (s SubscriptionSubjectIdentifier) NamespacedName() string {
+	return s.namespacedSubjectName[:strings.LastIndex(s.namespacedSubjectName, separator)]
+}
+
+// ConsumerName returns the JetStream consumer name.
+func (s SubscriptionSubjectIdentifier) ConsumerName() string {
+	return s.consumerName
+}
+
+// NewSubscriptionSubjectIdentifier returns a new SubscriptionSubjectIdentifier instance.
+func NewSubscriptionSubjectIdentifier(subscription *eventingv1alpha2.Subscription, subject string) SubscriptionSubjectIdentifier {
+	cn := computeConsumerName(subscription, subject)          // compute the consumer name once
+	nn := computeNamespacedSubjectName(subscription, subject) // compute the namespaced name with the subject once
+	return SubscriptionSubjectIdentifier{consumerName: cn, namespacedSubjectName: nn}
+}
+
+// computeConsumerName returns JetStream consumer name of the given subscription and subject.
+// It uses the crypto/md5 lib to return a string of 32 characters as recommended by the JetStream
+// documentation https://docs.nats.io/running-a-nats-service/nats_admin/jetstream_admin/naming.
+func computeConsumerName(subscription *eventingv1alpha2.Subscription, subject string) string {
+	cn := subscription.Namespace + separator + subscription.Name + separator + subject
+	h := md5.Sum([]byte(cn)) // #nosec
+	return hex.EncodeToString(h[:])
+}
+
+// computeNamespacedSubjectName returns Kubernetes namespaced name of the given subscription along with the subject.
+func computeNamespacedSubjectName(subscription *eventingv1alpha2.Subscription, subject string) string {
+	return subscription.Namespace + separator + subscription.Name + separator + subject
+}
+
+//----------------------------------------
+// CloudEvent utils
+//----------------------------------------
+
+func convertMsgToCE(msg *nats.Msg) (*cev2event.Event, error) {
+	event := cev2event.New(cev2event.CloudEventsVersionV1)
+	err := json.Unmarshal(msg.Data, &event)
+	if err != nil {
+		return nil, err
+	}
+	if err := event.Validate(); err != nil {
+		return nil, err
+	}
+	return &event, nil
+}
+
+//----------------------------------------
+// Testing utils
+//----------------------------------------
+
+func getCleanEventType(eventType string, cleaner cleaner.Cleaner) (string, error) {
```

Comment:
If this is only used in tests, then lets move it to testing pkg or in test file?

Reply:
Its actually used in the same file! I additionally wrote tests for it